src/Components/Web.JS/src/GlobalExports.ts

```diff
@@ -1,10 +1,12 @@
 import { navigateTo, internalFunctions as navigationManagerInternalFunctions } from './Services/NavigationManager';
 import { domFunctions } from './DomWrapper';
 import { Virtualize } from './Virtualize';
+import { registerCustomEventType } from './Rendering/Events/EventTypes';
 
 // Make the following APIs available in global scope for invocation from JS
 window['Blazor'] = {
   navigateTo,
+  registerCustomEventType,
```

Comment:
**API REVIEW** This is new public API on the JavaScript side (`Blazor.registerCustomEventType`). See the implementation at https://github.com/dotnet/aspnetcore/pull/29993/files#diff-58576c4c052d6fbfb851ec2f4b3bf740a2e318f0a9aeec9bc7aaf7a8acffe2eaR12 for details of the args.

I deliberately chose a hyperspecific name for this because (1) it's a pretty uncommon and advanced thing, and (2) I want to clarify that this is for registering the event *type*, and not something else like "event args mapping rule" which might be something else we add as a different feature in the future.

Reply:
Have you been thinking about how we would provide a typing (completion) experience as we continue to add newer JS APIs?

Reply:
Good point. We have talked about shipping a `.d.ts` file for the public APIs, but I wonder if some smaller change to the structure of `blazor.*.js` would be enough to make VS able to infer the exposed APIs and offer them based on the presence of the `<script src="blazor.*.js">` alone.

Reply:
I'm wondering if we can put a comment reference to the d.ts in `index.html` somehow to have editor's pick it up. we could put a reference to the tagged version for each release on our repo.